src/reconcile/rec_visibility.c

```diff
@@ -417,7 +417,11 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
     /* Should not see uncommitted changes in the history store */
     WT_ASSERT(session, !F_ISSET(S2BT(session), WT_BTREE_HS) || !list_uncommitted);
 
-    r->leave_dirty = r->leave_dirty || list_uncommitted;
+    if (list_uncommitted)
```

Comment:
Can you explain the difference between the logic of `leave_dirty` and `cache_write_restore`. How they will be used in which scenarios and what will be the effect in setting them here.

Reply:
leave_dirty is the page state after reconciliation. Both eviction and checkpoint can result to a page remain dirty after reconciliation. cache_write_store indicates whether we need to restore the saved updates to the new disk image after reconciliation. It should only happen in eviction. In short, it can be viewed as a replacement of the WT_REC_UPDATE_RESTORE flag.

Reply:
Can you add those details in a comment, so that it can be useful for while going through the code later?